src/protocol/graphics.ts

```diff
@@ -228,22 +228,31 @@ export function setupGraphics(store: UIStore) {
     });
   });
 
-  ThreadFront.on("paused", async () => {
-    if (!isRepaintEnabled()) {
-      return;
-    }
-
+  ThreadFront.on("paused", async ({ time }) => {
     await ThreadFront.ensureAllSources();
```

Comment:
Should this block the graphics stuff as well, or would it be worth doing like a `Promises.all` sort of thing to kick these off at the same time?

Reply:
Good point, I didn't think much about this part (because I didn't change it and because it's irrelevant once the recording is fully loaded) but it actually blocks the graphics stuff unnecessarily when the sources haven't been loaded yet (and that can take some time). I'll improve this.